### PR TITLE
[SHRINKRES-340] fix typos, grammar, errors and add missing descriptions in javadoc

### PR DESCRIPTION
**Pull request by xjusko**
https://issues.redhat.com/browse/SHRINKRES-340

### CODE REVIEW

File: api/src/main/java/org/jboss/shrinkwrap/resolver/api/Invokable.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Invokable {
      *
      * @param cl classloader to be used
      * @param classTypeName fully qualified class name
-     * @return
+     * @return               the loaded {@code Class} object
```
Comment (petrberan) [inline api/src/main/java/org/jboss/shrinkwrap/resolver/api/Invokable.java:59]: Why the extra space?

File: maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java [hunk 2]
```diff
@@ -36,11 +36,11 @@ class MavenCoordinateImpl extends MavenGABaseImpl implements MavenCoordinate {
      * required. If no {@link PackagingType} is specified, default will be to {@link PackagingType#JAR}. If no {@link ScopeType}
      * is specified, default will be {@link ScopeType#COMPILE}.
      *
-     * @param groupId
-     * @param artifactId
-     * @param version
-     * @param packaging
-     * @param classifier
+     * @param groupId     The group ID of the Maven coordinate.
```
Comment (petrberan) [inline maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java:39]: Why the extra space here?

File: maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java [hunk 3]
```diff
@@ -36,11 +36,11 @@ class MavenCoordinateImpl extends MavenGABaseImpl implements MavenCoordinate {
...
+     * @param artifactId  The artifact ID of the Maven coordinate.
+     * @param version     The version of the Maven coordinate.
+     * @param packaging   The packaging type of the Maven coordinate. Defaults to {@link PackagingType#JAR} if not specified.
+     * @param classifier The classifier of the Maven coordinate. If not specified, it defaults to the classifier derived from the packaging type.
```
Comment (petrberan) [inline maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java:43]: Here the extra space is missing

**Comment by xjusko** [comment]
Thanks @petrberan , I missed those when I was reviewing my changes, thanks, it should be fixed now.

File: maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java (hunk 2 above)
Comment (xjusko) [reply maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/coordinate/MavenCoordinateImpl.java:39]: It looks correct to me, so the descriptions are lined up for better readability in IDE

File: maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/integration/PomDependenciesUnitTestCase.java [hunk 4]
```diff
@@ -243,7 +243,7 @@ public void pomRemoteBasedDependencies() {
 
     /**
      * Tests resolving a version from pom.xml which has non-default scope
-     * https://issues.jboss.org/browse/SHRINKRES-103
+     * <a href="https://issues.jboss.org/browse/SHRINKRES-103">SHRINKRES-103</a>
```
Comment (petrberan) [inline maven/impl-maven/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/integration/PomDependenciesUnitTestCase.java:246]: See: <a href="https://issues.jboss.org/browse/SHRINKRES-103">SHRINKRES-103</a>

File: maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/strategy/DefaultTransitiveExclusionPolicy.java [hunk 5]
```diff
@@ -19,7 +19,7 @@
 import org.jboss.shrinkwrap.resolver.api.maven.ScopeType;
 
 /**
- * Applies the default behavior exhibited by Maven with regards to handling transitive dependencies during resolution.
+ * Applies the default behavior exhibited by Maven in regard to handling transitive dependencies during resolution.
```
Comment (petrberan) [inline maven/api-maven/src/main/java/org/jboss/shrinkwrap/resolver/api/maven/strategy/DefaultTransitiveExclusionPolicy.java:22]: What about "Applies the default behavior exhibited by Maven in handling transitive dependencies during resolution."

**Comment by petrberan** [comment]
I'm not a fan of those extra offset params in the javadoc. It may look good, but it's harder to maintain. Left some comments, otherwise looks good!

**Comment by petrberan** [comment]
Also please don't forget to link this PR to the JIRA issue @xjusko !

**Comment by xjusko** [comment]
@petrberan, The errors should be fixed, I also included changing all SHRINKRES jira links in Javadoc to HTML links. Regarding the indentation in Javadoc, I removed all the unnecessary whitespace as you suggested.

**Comment by petrberan** [comment]
Great job @xjusko , thank you

**Review by petrberan — APPROVED** [review]

**Pull request merged**